Packs/CortexAttackSurfaceManagement/Scripts/RankServiceOwners/RankServiceOwners.py

```diff
@@ -4,28 +4,131 @@
 from those surfaced by Cortex ASM Enrichment.
 """
 
-
-from typing import Any
-from collections.abc import Iterable
 import traceback
 from itertools import groupby
 import math
 
+import re
+import string
+import itertools
+import os
+import dill as pickle
+import posixpath
+import numpy as np
+import google.cloud.storage
+
+
+from typing import Any
+from collections.abc import Iterable, Callable
+
+
 STRING_DELIMITER = ' | '  # delimiter used for joining Source fields and any additional fields of type string
 
+# Normalize owner scores to be within the following bounds.
+# We want to use a standard scale (e.g. between 0 and 1) for interpretability.
+# However, we expect that normalizing to greater-than-half "probabilities" is
+# likely more accurate, # given that there are stringent conditions on initial detection
+# such that any name should be considered well-attested and likely to be an owner.
+SCORE_LOWER_BOUND = 0.5
+SCORE_UPPER_BOUND = 1.0
+
 
-def score(owners: list[dict[str, Any]]) -> list[dict[str, Any]]:
+def load_pickled_xpanse_object(file_name: str, cache_path: str = "/tmp/xpanse-ml") -> Any:
     """
-    Owner score is the number of observations on that owner divided by the max number of observations
-    for any owner in the list
+    Returns the pickled object at `file_name` as a Python object,
+    either using the local cache or retrieving from the
+    remote bucket as needed.
 
-    Expects `Count` key and replaces it with `Ranking Score`
+    The default cache is a subdirectory of /tmp directory will cache persistently across interactions.
+    Data saved to /var/lib/demisto will be lost betwen interactions (not cached).
     """
-    if owners:
-        max_count = max(owner.get('Count', 1) for owner in owners)
+    REMOTE_GCS_BUCKET = "xpanse-service-ownership-ml-models-dev"
+    REMOTE_GCS_PATH = ""  # ok for this to be empty string
+
+    os.makedirs(cache_path, exist_ok=True)
+    cached_file_path = os.path.join(cache_path, file_name)
+
+    # check that file is not empty; if authorization fails it will
+    # create the cache_path but the file will be empty
+    if not (os.path.exists(cached_file_path) and os.path.getsize(cached_file_path)):
+        remote_path = posixpath.join(REMOTE_GCS_PATH, file_name)
+
+        demisto.info(f"Starting download of '{file_name}' from gs://{REMOTE_GCS_BUCKET}/{remote_path}")
+        client = google.cloud.storage.client.Client()
```

Comment:
[Does this work without credentials?](https://cloud.google.com/python/docs/reference/storage/latest/google.cloud.storage.client.Client)

Reply:
^ great catch -- there should be a try/except around the appropriate code, with a meaningful log message if service ownership can't run (it may be all the way out in `main` or otherwise near the beginning of all this functionality)

Reply:
yes, the invocation is surrounded by a try/except which will assign a fixed fallback score per owner if the model can't be downloaded for any reason